implementation/src/main/java/io/smallrye/config/ConfigSourceInterceptor.java

```diff
@@ -32,4 +33,29 @@
      * @return a ConfigValue with information about the key name, value, config source origin and ordinal.
      */
     ConfigValue getValue(ConfigSourceInterceptorContext context, String name);
+
+    default Iterable<String> getNames(ConfigSourceInterceptorContext context) {
```

Comment:
Streams are significantly more difficult to intercept AFAICT.  You're limited to a much smaller number of operations.  Streams also are likelyt to perform worse with many more allocations.

Reply:
I would use `Iterator` TBH.  It's one less object to construct.

Reply:
I had Iterator before but Config uses Iterable so I've changed it to be consistent. What do you think?

Reply:
Config is the front end interface; its `Iterable` can trivially use the interceptor chain's `Iterator` to materialize it.  It's better to do this at the front end because otherwise every interceptor that needs to wrap will have to wrap both the `Iterable` and the `Iterator` that it creates.

Reply:
Ok. I'll change that.